src/write.c

```diff
@@ -1282,10 +1299,8 @@ avifResult avifEncoderFinish(avifEncoder * encoder, avifRWData * output)
         return AVIF_RESULT_NO_CONTENT;
     }
 
-    // TODO(yguyon): Rework when AVIF_CODEC_CHOICE_AUTO can be AVM
-    assert((encoder->codecChoice != AVIF_CODEC_CHOICE_AUTO) ||
-           (strcmp(avifCodecName(encoder->codecChoice, AVIF_CODEC_FLAG_CAN_ENCODE), "avm") != 0));
-    const avifBool isAV2 = encoder->codecChoice == AVIF_CODEC_CHOICE_AVM;
+    const avifCodecType codecType = avifEncoderGetCodecType(encoder);
+    AVIF_CHECKERR(codecType != AVIF_CODEC_TYPE_UNKNOWN, AVIF_RESULT_NO_CODEC_AVAILABLE);
```

Comment:
can you please change this to a normal if condition?

Use of AVIF_CHECKERR makes this a double negation and difficult to parse.